src/features/ProtectTransaction/ProtectTxProvider.tsx

```diff
@@ -209,6 +219,22 @@ const ProtectTxProvider: React.FC = ({ children }) => {
     [setState]
   );
 
+  const setFeeAmount = useCallback(
+    async (feeAmount: IFeeAmount) => {
+      if (!feeAmount) {
+        return Promise.reject();
+      }
+
+      setState((prevState) => ({
+        ...prevState,
+        feeAmount
+      }));
+
+      return Promise.resolve();
+    },
+    [setState]
+  );
```

Comment:
1. This Promise gives the false impression that the caller can expect the state to be updated on resolve. Please remove.
2. `setState` is never updated. Remove from deps.